monitoring/backups_collector.go

```diff
@@ -0,0 +1,94 @@
+package monitoring
+
+import (
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+)
+
+type BackupsCollector struct {
+	*baseCollector
+
+	sizeMetric  metricInfo
+	stateMetric metricInfo
+}
+
+func NewBackupsCollector(
+	logger logrus.FieldLogger,
+	nodeID string,
+	ds *datastore.DataStore) *BackupsCollector {
+
+	vc := &BackupsCollector{
+		baseCollector: newBaseCollector(subsystemBackups, logger, nodeID, ds),
+	}
+
+	vc.sizeMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemBackups, "actual_size_bytes"),
+			"Actual space used by each backup of the snapshot on the corresponding node",
+			[]string{nodeLabel, backupsLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	vc.stateMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemBackups, "stats_backup_status"),
+			"State of this backup",
+			[]string{nodeLabel, backupsLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	return vc
+}
+
+func (vc *BackupsCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- vc.sizeMetric.Desc
+	ch <- vc.stateMetric.Desc
+}
+
+func (vc *BackupsCollector) Collect(ch chan<- prometheus.Metric) {
+	defer func() {
+		if err := recover(); err != nil {
+			vc.logger.WithField("error", err).Warn("panic during collecting metrics")
+		}
+	}()
+
+	backupsLists, err := vc.ds.ListBackups()
```

Comment:
nit: it'd be good to add a new function `ListBackupsRO` inside the datastore function to avoid the DeepCopy all the backups.

Reply:
Resolved on new commit https://github.com/longhorn/longhorn-manager/commit/ba6d2ee5add07fe03aae932d0c9423ed1a8e67c8